src/fonts/monospaced_pixels.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{
+    fonts::{MonoFont, Text},
+    geometry::Point,
+    pixelcolor::PixelColor,
+    style::MonoTextStyle,
+    Pixel,
+};
+
+/// Pixel iterator for styled text with a monospaced font.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub struct MonoPixels<'a, C, F>
+where
+    C: PixelColor,
+    F: MonoFont,
+{
+    char_walk_x: i32,
+    char_walk_y: i32,
+    current_char: Option<char>,
+    idx: usize,
+    top_left: Point,
+    pos: Point,
+    text: &'a str,
+    style: MonoTextStyle<C, F>,
+}
+
+impl<'a, C, F> MonoPixels<'a, C, F>
+where
+    C: PixelColor,
+    F: MonoFont,
+{
+    pub(crate) fn new(text: &Text<'a>, style: MonoTextStyle<C, F>) -> Self {
+        Self {
+            current_char: text.text.chars().next(),
+            idx: 0,
+            text: text.text,
+            char_walk_x: 0,
+            char_walk_y: 0,
+            top_left: text.position,
+            pos: text.position,
+            style,
+        }
+    }
+}
+
+impl<C, F> Iterator for MonoPixels<'_, C, F>
+where
+    C: PixelColor,
+    F: MonoFont,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            if self.current_char == Some('\n') {
```

Comment:
Thanks for keeping this in :)

Reply:
Sorry, but the final verdict is still open on this one. After this PR each font renderer still needs to handle newlines itself, but I'll look into it after this is merged.

Reply:
I think that's ok for external fonts. I'd like to keep at as a feature of the builtin mono fonts if it's not too complicated. Otherwise I'll concede and just use multiple `Text` calls :)

Reply:
> I think that's ok for external fonts.

I don't ;). I would like to keep the external text renderer API as simple as possible.

> Otherwise I'll concede and just use multiple `Text` calls :)

I don't think you need to. It should be possible to move the handling of newlines from the text renderers to the styled text drawable. This way each text renderer would only need to render single lines and get the the multiline support for free. But I'll investigate this in another PR.

Reply:
Sounds good!